src/udev/net/link-config.c

```diff
@@ -375,13 +375,21 @@ int link_config_apply(link_config_ctx *ctx, link_config *config,
                 if (config->port != _NET_DEV_PORT_INVALID)
                         log_warning_errno(r,  "Could not set port (%s) of %s: %m", port_to_string(config->port), old_name);
 
-                speed = DIV_ROUND_UP(config->speed, 1000000);
-                if (r == -EOPNOTSUPP)
-                        r = ethtool_set_speed(&ctx->ethtool_fd, old_name, speed, config->duplex);
+                if (config->advertise)
+                        log_warning_errno(r, "Could not set advertise to %s: %m", advertise_to_string(config->advertise));
```

Comment:
done

Reply:
What do you mean by "done"?

Try the following code:
```C
        log_warning("Could not set advertise to %s: %m",
                    advertise_to_string(NET_DEV_ADVERTISE_100BASET_HALF));
        log_warning("Could not set advertise to %s: %m",   
              advertise_to_string(NET_DEV_ADVERTISE_100BASET_HALF|
              NET_DEV_ADVERTISE_10BASET_HALF));
```
This prints
```
Could not set advertise to 100baset-half: Success
Could not set advertise to (null): Success
```
... because in the first case, there's just a single bit set, but in the second two.